ykrt/src/mt.rs

```diff
@@ -1319,17 +1324,18 @@ impl MTThread {
 
     /// Is this thread currently tracing something?
     pub(crate) fn is_tracing() -> bool {
-        THREAD_IS_TRACING.with(|x| x.load(Ordering::Relaxed) != IsTracing::None)
+        unsafe { __yk_thread_tracing_state != IsTracing::None }
     }
 
     /// What kind of tracing (if any!) is this thread undertaking?
     fn tracing_kind() -> IsTracing {
-        THREAD_IS_TRACING.with(|x| x.load(Ordering::Relaxed))
+        let raw = &raw mut __yk_thread_tracing_state;
```

Comment:
I will try again (in a follow up PR) to use an atomic here, thus avoiding the need for &raw, but as you can see, this PR is already big enough.